adapters/handlers/graphql/local/common_filters/extract_targets.go

```diff
@@ -0,0 +1,100 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package common_filters
+
+import (
+	"fmt"
+
+	"github.com/weaviate/weaviate/entities/dto"
+)
+
+func ExtractTargets(source map[string]interface{}) ([]string, *dto.TargetCombination, error) {
```

Comment:
I see this can fail with an error, but it seems like the actual usage in the code ignores these. Would it be better to not let this silently fail?

Reply:
The problem is that the nearText etc are handled by the module system and only return interface{} and the whole thing is quite convoluted :( - the error is handled in the other branches with direct returns (nearVector etc)

I added another commit that refactors the return system. Not sure if it is worth it. Note that there is simply no error handling for the other parts of the code

Reply:
will merge this, let me know if you want me to change anything :)